lib/nostrum/util.ex

```diff
@@ -152,9 +152,14 @@ defmodule Nostrum.Util do
 
   defp get_new_gateway_url do
     case Api.request(:get, Constants.gateway_bot(), "") do
+      # Handle deeper API error.
+      # For example, not including a token can result in a somewhat opaque 401 unauthorized error
+      {:error, %Nostrum.Error.ApiError{response: %{message: message}}} ->
+        raise(message)
+
       {:error, %{status_code: code, message: message}} ->
```

Comment:
Like I said in the Discord channel this is out of date in regards to the [new](https://github.com/Kraigie/nostrum/blob/master/lib/nostrum/api/ratelimiter.ex#L154) return formats. I believe it can be built into the other error case.